### PR TITLE
Update auth links with correct settings for multisite

### PR DESCRIPTION
**Pull request by codiebeulaine**
We need to update the auth service links with the correct parameters dynamically based on the `request.site`.

I'm not sure if this is the best way to do it, we could keep these links in the settings file and use middleware to update it as an alternative? but these values are only called in the templates and bad practice to use middleware to update settings?

Any thoughts?

### CODE REVIEW

File: gem/context_processors.py [hunk 1]
```diff
@@ -33,12 +33,39 @@ def detect_freebasics(request):
...
+            VIEW_PROFILE_URL = (
+                "%s/profile/edit/?theme=%s&redirect_uri=%s&client_id=%s" % (
+                    settings.OIDC_OP, settings.THEME,
+                    oidc_settings.wagtail_redirect_url,
+                    oidc_settings.oidc_rp_client_id))
+            EDIT_PROFILE_URL = (
+                "%s/profile/edit/?theme=%s&redirect_uri=%s&client_id=%s" % (
+                    settings.OIDC_OP, settings.THEME,
+                    oidc_settings.wagtail_redirect_url,
+                    oidc_settings.oidc_rp_client_id))
```
Comment (sewagodimo) [inline gem/context_processors.py]: what about log out? 
and login?
Are we using the oidc urls for that?

File: gem/settings/base.py [hunk 2]
```diff
@@ -63,15 +63,6 @@
 if USE_OIDC_AUTHENTICATION:
```
Comment (sewagodimo) [inline gem/settings/base.py]: what about passing the theme to the login url or should we do that in another PR??

File: gem/context_processors.py [hunk 3]
```diff
@@ -33,12 +33,39 @@ def detect_freebasics(request):
...
+        if not hasattr(site, "oidcsettings"):
+            raise RuntimeError(
+                "Site {} has no settings configured.".format(site))
+
+        oidc_settings = site.oidcsettings
+        if oidc_settings:
+            REGISTRATION_URL = (
+                "%s/registration/?theme=%s&hide=end-user&redirect_uri=%s" % (
+                    settings.OIDC_OP, settings.THEME,
+                    oidc_settings.wagtail_redirect_url))
```
Comment (sewagodimo) [inline gem/context_processors.py]: should registration not be redirecting to the home page?

Comment (codiebeulaine) [reply gem/context_processors.py]: @sewagodimo  the wagtail redirect url can be set to the homepage in the OIDC settings

File: gem/settings/base.py (hunk 2 above)
Comment (codiebeulaine) [reply gem/settings/base.py]: Did that in a separate PR, it is handled differently to everything else
